version.go

```diff
@@ -1,19 +1,27 @@
 package main
 
 import (
+	_ "embed"
+	"strings"
+
 	goversion "github.com/hashicorp/go-version"
 )
 
-// The main version number that is being run at the moment.
-var version = "0.0.0"
+var (
+	// The next version number that will be released. This will be updated after every release
+	// Version must conform to the format expected by github.com/hashicorp/go-version
+	// for tests to work.
+	// A pre-release marker for the version can also be specified (e.g -dev). If this is omitted
+	// then it means that it is a final release. Otherwise, this is a pre-release
+	// such as "dev" (in development), "beta", "rc1", etc.
+	//go:embed version/VERSION
+	fullVersion string
 
-// A pre-release marker for the version. If this is "" (empty string)
-// then it means that it is a final release. Otherwise, this is a pre-release
-// such as "dev" (in development), "beta", "rc1", etc.
-var versionPrerelease = "dev"
+	version, versionPrerelease, _ = strings.Cut(fullVersion, "-")
```

Comment:
Nitpicky idea: Could we possibly call a custom function which leverages `go-version` and returns the different parts, instead of re-implementing small part of it via `strings.Cut` which may not cover all edge cases correctly? e.g. `1.2.3+meta-data`.

For example
```go
func parseRawVersion(rawVersion string) (version, prerelease, meta string) {
	v := goversion.Must(goversion.NewVersion(rawVersion))
	version = v.Core().String()
	prerelease = v.Prerelease()
	meta = v.Metadata()
}
```

We _might_ be able to then also omit the `init()` check, but I'm not sure how `panic`s are handled in this context - that would need double checking.

Reply:
Manually calling `panic` inside `init` or using `goversion.Must` inside a `parseRawVersion` function produces similar results. Any `terraform-ls` command will panic:

```
$ terraform-ls serve -port 3333
panic: Malformed version: 1.0.0-beta+exp.sha.5114f85"

goroutine 1 [running]:
github.com/hashicorp/go-version.Must(...)
        /Users/dbanck/go/pkg/mod/github.com/hashicorp/go-version@v1.6.0/version.go:102
main.parseRawVersion({0x17733b6?, 0x1a0?})
        /Users/dbanck/Projects/terraform-ls/version.go:28 +0x8e
main.init()
        /Users/dbanck/Projects/terraform-ls/version.go:19 +0x39
```

Instead of storing version, prerelease and metadata in separate variables, we could store the `Version` instead. So we don't need to recreate the version string manually and can return `version.String()`. What do you think of this approach?